### PR TITLE
feat(tycho-client-py): Add a tycho-client python package.

### PR DESCRIPTION
**Pull request by kayibal**
This commit adds support for connecting to a tycho stream of block synchronized protocol updates.

### CODE REVIEW

File: tycho-client-py/tycho_client/dto.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    creation = "Creation"
+    unspecified = "Unspecified"
+
+
+class ExtractorIdentity(BaseModel):
+    chain: Chain
+    name: str
+
+    @classmethod
+    def new(cls, chain: Chain, name: str) -> "ExtractorIdentity":
```
Comment (tvinagre) [inline tycho-client-py/tycho_client/dto.py]: I'm curious why did you create this if it's already a Pydantic Model?

File: tycho-client-py/tycho_client/stream.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+from pydantic import ValidationError
+
+from .dto import Chain, FeedMessage
+from .exception import TychoStreamException
+
+log = getLogger(__name__)
+
+
+class TychoStream:
+    # we allow to initialise this class with some preset PoolStates for use cases such as DAIPSM that do not need
```
Comment (tvinagre) [inline tycho-client-py/tycho_client/stream.py]: I think this comment is not needed here - or we need to add back the `pool_state` parameter.

File: tycho-client-py/tycho_client/dto.py (hunk 1 above)
Comment (kayibal) [reply tycho-client-py/tycho_client/dto.py]: Uhps that was chatgpt

File: tycho-client-py/tycho_client/rpc_client.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+        res = self._post_request(f"/v1/{self._chain}/tokens", body=body, params={})
+        return [ResponseToken(**t) for t in res["tokens"]]
+
+
+if __name__ == "__main__":
+    client = TychoRPCClient("http://localhost:4242")
+    print(client.get_protocol_components("uniswap_v2"))
+    print(client.get_protocol_state(protocol_system="vm:balancer"))
+    print(client.get_contract_state(include_balances=True))
```
Comment (zizou0x) [inline tycho-client-py/tycho_client/rpc_client.py]: This looks like a test logic, should it stay?

Comment (kayibal) [reply tycho-client-py/tycho_client/rpc_client.py]: Probably useful to quickly check tings

**Pull request merged**